src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringPipelineListener.java

```diff
@@ -135,6 +139,39 @@ public void onStartStageStep(@Nonnull StepStartNode stepStartNode, @Nonnull Stri
         }
     }
 
+    @Override
+    public void onStartCreateSpanStep(@Nonnull StepStartNode stepStartNode, @Nonnull WorkflowRun run) {
+        try (Scope ignored = setupContext(run, stepStartNode)) {
+            verifyNotNull(ignored, "%s - No span found for node %s", run, stepStartNode);
+
+            String createSpanName = getStepName(stepStartNode.getDescriptor(), "createSpan");
+            String stepType = getStepType(stepStartNode.getDescriptor(), "step");
+            JenkinsOpenTelemetryPluginConfiguration.StepPlugin stepPlugin = JenkinsOpenTelemetryPluginConfiguration.get().findStepPluginOrDefault(stepType, stepStartNode);
+
+            final Map<String, Object> arguments = ArgumentsAction.getFilteredArguments(stepStartNode);
+            final String spanAttribute = (String) arguments.getOrDefault("name", createSpanName);
```

Comment:
I don't understand the word `attribute` in the variable name. I thought it would be a span attribute but in fact it seems to be an attribute in the sense of an attribute of the `createSpan` step of the Jenkins Pipeline DSL. IS it correct?

Reply:
Yes, I'm renaming the variable to avoid the misleading